src/ui/button/button.tsx

```diff
@@ -1,29 +1,19 @@
-import { styled } from "styled-components";
-
-type Props = {
-    children: string;
-    disabled?: boolean;
-    onClick: () => void;
-};
-
-export const Button: React.FC<Props> = ({children, disabled, onClick}) =>{
-    return  <ButtonWrapper type="button"disabled={disabled} onClick={()=> onClick}>
-        {children}
-    </ButtonWrapper>       
-};
-
-const ButtonWrapper = styled.button`
-all:unset;
-cursor: pointer;
-border: 1px solid white;
-border-radius: 4px;
-padding: 5px 10px;
-background-color: blue;
-color: white;
-
-&:disabled {
-    cursor: none;
-    opacity: 0.67;
-    color: grey;
-}
-`;
\ No newline at end of file
+import { ButtonWrapper } from './button.styles';
+
+type MyFunction = () => void;
+
+type ButtonProps = {
+  children: string;
+  disabled?: boolean;
+  onClick: MyFunction;
+};
+
+export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
+  return (
+    <ButtonWrapper type="button" disabled={props.disabled} onClick={props.onClick}>

```

Comment:
when adding a listener to html element, lambda style is preferable:

```typescript
<ButtonWrapper type="button" disabled={props.disabled} onClick={() => props.onClick()}>
```

because onClick emits Event so it can be used by parent logic unusually